src/scf-release/src/acceptance-tests-brain/test-scripts/010_nfspersi_test.sh

```diff
@@ -0,0 +1,176 @@
+    do
+	sleep 10
+    done
+}
+
+# Replace the placeholder storage class for persistent volumes with
+# the actual class provided by the execution environment.
+SKUBEC=${SELFDIR}/../test-resources/nfs_server_kube.yaml
+sed -e "s/storage-class: \"persistent\"/storage-class: \"${SC}\"/" \
+    < "${SKUBEC}" \
+    > "${TMP}/nfs_server_kube.yaml"
+SKUBEC="${TMP}/nfs_server_kube.yaml"
+
+kubectl create -n "${NS}" -f "${SKUBEC}"
+wait "${NS}"
+
+# Server of the NFS volume to use, as name (pulled from the kube config)
+SNAME="$(grep '^  name:' "${SKUBEC}" | awk '{print $2}' | sed -e "s/\"//g")"
+
+# Server of the NFS volume to use, as IP address (pulled from kube runtime via name)
+SADDR="$(kubectl describe service -n "${NS}" "${SNAME}" | grep ^IP | awk '{print $2}')"
+
+# Now that we have an NFS server, with an exportable volume, we can
+# configure it for actual export.
+##
+# - Permissions for all
+# - Declare as NFS export (insecure = allow any origin port)
+# - Update the NFS master tables to include the new export
+
+echo > "${TMP}/export" '/exports/foo *(rw,insecure)'
+kubectl cp "${TMP}/export" "${NS}/${SNAME}-0:/etc/exports.d/foo.exports"
+
+kubectl exec -n "${NS}" "${SNAME}-0" -- chmod a+rwx /exports/foo
+kubectl exec -n "${NS}" "${SNAME}-0" -- exportfs -a
+
+# Fix IP in the various configuration files
+sed < ${SGROUP} > ${TMP}/nfs_secgroup.json  -e "s|192.168.77.77|${SADDR}|"
+sed < ${SMOUNT} > ${TMP}/nfs_mount.json     -e "s|192.168.77.77|${SADDR}|"
+
+SGROUP="${TMP}/nfs_secgroup.json"
+SMOUNT="${TMP}/nfs_mount.json"
+
+# Show the security group, for debugging.
+echo ======================== ; cat "${SGROUP}" ; echo ========================
+echo ======================== ; cat "${SMOUNT}" ; echo ========================
+
+# Create a security group which allows access to the nfs server
+# Deploy the pora app of the pats, and bind it to a persi service
+
+cf create-security-group       nfs-test ${SGROUP}
+cf bind-running-security-group nfs-test
+cf bind-staging-security-group nfs-test
+
+cd $PORAPP
+cf push pora --no-start
+
+cf enable-service-access persi-nfs
+cf create-service        persi-nfs Existing myVolume -c "$(cat "${SMOUNT}")"
+
+cf bind-service pora myVolume -c '{"uid":"1000","gid":"1000"}'
+cf start pora
+
+APP=pora.${CF_DOMAIN}
+PATTERN='Hello Persistent World!'
+
+# Test that the app is available
+curl "${APP}"
+
+# Test that the app can write to the volume of the bound service
+curl "${APP}/write" | grep "${PATTERN}"
+
+# Test that we can create, read, chmod, and delete a file.  We check
+# the curl results in part, and, more importantly the pod holding the
+# actual volume in part.
+
+FNAME=$(curl "${APP}/create")
+
+kubectl exec -n "${NS}" "${SNAME}-0" -- ls /exports/foo | grep "${FNAME}"
+kubectl exec -n "${NS}" "${SNAME}-0" -- cat "/exports/foo/${FNAME}" | grep "${PATTERN}"
```

Comment:
Could `grep "${PATTERN}" "/exports/foo/${FNAME}"` be used here? If not, can you add a comment why?  My guess is that it would be related to exit status, although I think it should work.

Reply:
It might be. I do not remember anymore what I was thinking.

Can you test ? If I test it will take a while before you have results, as I have to make me a vagrant box first.

Reply:
I confirmed that it works - also confirmed it failed if I changed it to grep for something other than `${PATTERN}`

```
--- src/scf-release/src/acceptance-tests-brain/test-scripts/010_nfspersi_test.sh
+++ src/scf-release/src/acceptance-tests-brain/test-scripts/010_nfspersi_test.sh
@@ -166,7 +166,7 @@ curl "${APP}/write" | grep "${PATTERN}"
 FNAME=$(curl "${APP}/create")

 kubectl exec -n "${NS}" "${SNAME}-0" -- ls /exports/foo | grep "${FNAME}"
-kubectl exec -n "${NS}" "${SNAME}-0" -- cat "/exports/foo/${FNAME}" | grep "${PATTERN}"
+kubectl exec -n "${NS}" "${SNAME}-0" -- grep "${PATTERN}" "/exports/foo/${FNAME}"

 curl "${APP}/read/${FNAME}" | grep "${PATTERN}"
```

Reply:
Thanks. Integrating it now, then look into the cleanup issues.